packages/dd-trace/src/appsec/iast/analyzers/sql-injection-analyzer.js

```diff
@@ -20,37 +20,33 @@ class SqlInjectionAnalyzer extends InjectionAnalyzer {
     this.addSub('apm:mysql2:query:start', ({ sql }) => this.analyze(sql, 'MYSQL'))
     this.addSub('apm:pg:query:start', ({ query }) => this.analyze(query.text, 'POSTGRES'))
 
-    this.addSub('datadog:sequelize:query:start', ({ sql, dialect }) => {
-      const parentStore = storage.getStore()
-      if (parentStore) {
-        this.analyze(sql, dialect.toUpperCase(), parentStore)
-
-        storage.enterWith({ ...parentStore, sqlAnalyzed: true, sequelizeParentStore: parentStore })
-      }
-    })
-
-    this.addSub('datadog:sequelize:query:finish', () => {
-      const store = storage.getStore()
-      if (store && store.sequelizeParentStore) {
-        storage.enterWith(store.sequelizeParentStore)
-      }
-    })
-
-    this.addSub('datadog:pg:pool:query:start', ({ query }) => {
-      const parentStore = storage.getStore()
-      if (parentStore) {
-        this.analyze(query.text, 'POSTGRES', parentStore)
-
-        storage.enterWith({ ...parentStore, sqlAnalyzed: true, pgPoolParentStore: parentStore })
-      }
-    })
-
-    this.addSub('datadog:pg:pool:query:finish', () => {
-      const store = storage.getStore()
-      if (store && store.pgPoolParentStore) {
-        storage.enterWith(store.pgPoolParentStore)
-      }
-    })
+    this.addSub(
+      'datadog:sequelize:query:start',
+      ({ sql, dialect }) => this.getStoreAndAnalyze(sql, dialect.toUpperCase())
+    )
+    this.addSub('datadog:sequelize:query:finish', this.returnToParentStore)
```

Comment:
With the current approach, youare replacing the `this` property of `returnToParentStore` to unknown one.
```suggestion
    this.addSub('datadog:sequelize:query:finish', () => this.returnToParentStore())
```

Reply:
same in L33